src/main/java/duke/tool/Storage.java

```diff
@@ -0,0 +1,83 @@
+package duke.tool;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import duke.tasks.Deadline;
+import duke.tasks.Event;
+import duke.tasks.Task;
+import duke.tasks.Todo;
+
+/**
+ * The class deals with loading tasks from the file and saving tasks in the file.
+ */
+public class Storage {
+    private String filePath;
+
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Load the data in the file to the task list.
+     * @return
+     * @throws IOException
+     */
+    public ArrayList<Task> loadData() throws IOException {
+        DateTimeFormatter validFormat = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");
+        ArrayList<Task> orderList = new ArrayList<>();
+
+        try {
+            File dataStorage = new File(filePath);
+            Scanner s = new Scanner(dataStorage);
+            while (s.hasNext()) {
+                String curr = s.nextLine();
+                String[] currTask = curr.split("|");
+                Boolean isDone = currTask[1] == "1";
+                if (currTask[0] == "T") {
```

Comment:
This could also be another place where you might want to consider using a switch-case loop (=

Reply:
For example,
switch(currTask[0]) {
case("T") :
    do something
}